### PR TITLE
Simplify function score API

### PR DESCRIPTION
**Pull request by kharnt0x**
simplifying API as per https://github.com/sksamuel/elastic4s/issues/1047
Also added tests for function_score query as a whole when using TCP client

### CODE REVIEW

File: elastic4s-core/src/main/scala/com/sksamuel/elastic4s/searches/queries/funcscorer/FunctionScoreQueryDefinition.scala [hunk 1]
```diff
@@ -29,10 +29,10 @@ case class FunctionScoreQueryDefinition(query: Option[QueryDefinition] = None,
               rest: ScoreFunctionDefinition*): FunctionScoreQueryDefinition = scorers(first +: rest)
 
   def scorers(scorers: Iterable[ScoreFunctionDefinition]): FunctionScoreQueryDefinition =
-    scoreFuncs(scorers.map(FilterFunctionDefinition(_)))
+    scoreFuncs(scorers)
 
-  def scoreFuncs(first: FilterFunctionDefinition,
```
Comment (kharnt0x) [inline elastic4s-core/src/main/scala/com/sksamuel/elastic4s/searches/queries/funcscorer/FunctionScoreQueryDefinition.scala]: @sksamuel  we could remove `scoreFuncs` altogether now and just leave `scorers`

**Comment by sksamuel** [comment]
Overall very good. One comment - should we rename scorers to functions, since it's called functions in the json and that's what people will see when they read the elasticsearch docs. Then we can remove scoreFuncs as you say as well. It was called scorers in the code as a left over from the Java TCP API. 

Also, `scriptScore` is a proper field in the `FunctionScoreQueryDefinition` but it could be removed as it's a score function anyway.

**Comment by kharnt0x** [comment]
@sksamuel good suggestions, this should help with discoverability for sure. I've made the updates

**Comment by sksamuel** [comment]
Sorry to be a pain, can we keep the original methods just with deprecation. I'll merge this if you want to do that as a separate pr.

**Pull request merged**